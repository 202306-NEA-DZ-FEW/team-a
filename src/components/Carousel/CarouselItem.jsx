import Link from "next/link";

export default function CarouselItem({
  t,
  name,
  dataKey,
  imageURL,
  description,
  queryParams,
}) {
  return (
    <div
      className='item w-20 h-20 lg:w-[200px] lg:h-1/2 shadow-2xl absolute z-10 lg:top-[50%] top-[80%] lg:-translate-y-[40%] lg:translate-x-[-30%] -translate-x-[170%] bg-blend-overlay bg-black bg-opacity-20'
      style={{ backgroundImage: `url(${imageURL})` }}
    >
      <div className='content absolute z-10 top-[50%] left-[5%] lg:max-w-md text-white -translate-y-[50%] hidden'>
        <h1 className='text-4xl font-bold my-4 name'>{name}</h1>
        <p className='text-lg font-light mb-4 des'>
          {description ? description : ""}
        </p>
        <Link
          className='btn btn-sm bg-white text-black hover:bg-slate-50'
          scroll={false}
          href={{
            pathname: "/products",
            query: { ...queryParams, category: dataKey },
          }}
        >
          {t("productsPage:seeMore")}
        </Link>
      </div>
    </div>
  );
}
