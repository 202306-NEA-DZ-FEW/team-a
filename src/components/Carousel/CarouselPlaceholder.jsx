function CarouselPlaceholder() {
  return (
    <div className='w-full h-full flex items-center justify-center'>
      <div className='w-[95%] p-8 flex flex-col lg:flex-row justify-between h-[90%] animate-pulse mx-auto rounded-2xl'>
        <div className='lg:w-[40%] w-full mt-20'>
          <div className='w-[80%] h-[50px] bg-gray-300 animate-pulse rounded-full mt-20'></div>
          <div className='w-[100px] h-[40px] bg-gray-300 animate-pulse rounded-full mt-4'></div>
        </div>
        <div className='flex gap-4 mt-auto lg:w-[60%] w-full items-center justify-center'>
          <div className='lg:w-[50%] lg:h-[320px] w-[50px] h-[50px] bg-gray-300 animate-pulse rounded-xl mt-4'></div>
          <div className='lg:w-[50%] lg:h-[320px] w-[50px] h-[50px] bg-gray-300 animate-pulse rounded-xl mt-4'></div>
          <div className='lg:w-[50%] lg:h-[320px] w-[50px] h-[50px] bg-gray-300 animate-pulse rounded-xl mt-4'></div>
        </div>
      </div>
    </div>
  );
}

export default CarouselPlaceholder;
