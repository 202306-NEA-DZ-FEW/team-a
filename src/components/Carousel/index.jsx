import { MdChevronLeft, MdChevronRight } from "react-icons/md";

import CarouselItem from "./CarouselItem";

export default function Carousel({ items, queryParams }) {
  const nextItem = () => {
    let lists = document.querySelectorAll(".item");
    document.getElementById("slide").appendChild(lists[0]);
  };

  const prevItem = () => {
    let lists = document.querySelectorAll(".item");
    document.getElementById("slide").prepend(lists[lists.length - 1]);
  };

  return (
    <div className='group'>
      <div id='slide' className='max-w-fit'>
        {items.map((item, i) => (
          <CarouselItem
            key={item.dataKey + i}
            {...item}
            queryParams={queryParams}
          />
        ))}
      </div>
      <div className='absolute px-8 z-50 lg:bottom-12 bottom-[10%] justify-between w-full flex gap-4 lg:justify-center'>
        <button
          className='btn btn-circle bg-white hover:bg-slate-50 border-none items-center justify-center lg:hidden lg:group-hover:flex'
          onClick={prevItem}
        >
          <MdChevronLeft size={40} />
        </button>
        <button
          className='btn btn-circle bg-white hover:bg-slate-50 border-none items-center justify-center lg:hidden lg:group-hover:flex'
          onClick={nextItem}
        >
          <MdChevronRight size={40} />
        </button>
      </div>
    </div>
  );
}
