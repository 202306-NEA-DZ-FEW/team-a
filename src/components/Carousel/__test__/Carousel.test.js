import image from "public/images/mepi1.png";
import renderer from "react-test-renderer";

import Carousel from "..";

it("renders correctly", () => {
  const mockItems = [
    {
      id: 0,
      name: "name",
      description: "description",
      dataKey: "dataKey",
      imageURL: image,
    },
  ];
  const queryParams = {};
  const mockT = jest.fn();
  const tree = renderer
    .create(<Carousel items={mockItems} queryParams={queryParams} t={mockT} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
