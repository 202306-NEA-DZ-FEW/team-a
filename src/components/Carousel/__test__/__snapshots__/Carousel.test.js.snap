// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="group"
>
  <div
    className="max-w-fit"
    id="slide"
  >
    <div
      className="item w-20 h-20 lg:w-[200px] lg:h-1/2 shadow-2xl absolute z-10 lg:top-[50%] top-[80%] lg:-translate-y-[40%] lg:translate-x-[-30%] -translate-x-[170%] bg-blend-overlay bg-black bg-opacity-20"
      style={
        Object {
          "backgroundImage": "url([object Object])",
        }
      }
    >
      <div
        className="content absolute z-10 top-[50%] left-[5%] lg:max-w-md text-white -translate-y-[50%] hidden"
      >
        <h1
          className="text-4xl font-bold my-4 name"
        >
          name
        </h1>
        <p
          className="text-lg font-light mb-4 des"
        >
          description
        </p>
        <a
          className="btn btn-sm bg-white text-black hover:bg-slate-50"
          href="/products?category=dataKey"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          See more
        </a>
      </div>
    </div>
  </div>
  <div
    className="absolute px-8 z-50 lg:bottom-12 bottom-[10%] justify-between w-full flex gap-4 lg:justify-center"
  >
    <button
      className="btn btn-circle bg-white hover:bg-slate-50 border-none items-center justify-center lg:hidden lg:group-hover:flex"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height={40}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={40}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
    </button>
    <button
      className="btn btn-circle bg-white hover:bg-slate-50 border-none items-center justify-center lg:hidden lg:group-hover:flex"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height={40}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={40}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </button>
  </div>
</div>
`;
